File: apps/api/app/schemas/wellness.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict

from app.models.wellness_score import BurnoutRiskLevel


class WellnessScoreResponse(BaseModel):
    id: uuid.UUID
    score_date: date
    composite_score: int | None
    mood_component: int | None
    sleep_component: int | None
    stress_component: int | None
    energy_component: int | None
    activity_component: int | None
    journaling_component: int | None
    burnout_risk_score: int | None
    burnout_risk_level: BurnoutRiskLevel | None

    model_config = ConfigDict(from_attributes=True)

File: apps/web/src/lib/api/wellness.ts
import { apiFetch } from "./client";

export type BurnoutRiskLevel = "green" | "amber" | "red";

export interface WellnessScoreResponse {
  id: string;
  score_date: string;
  composite_score: number | null;
  mood_component: number | null;
  sleep_component: number | null;
  stress_component: number | null;
  energy_component: number | null;
  activity_component: number | null;
  journaling_component: number | null;
  burnout_risk_score: number | null;
  burnout_risk_level: BurnoutRiskLevel | null;
}

export async function getWellnessScores(days: number = 30): Promise<WellnessScoreResponse[]> {
  const res = await apiFetch(`/wellness/score?days=${days}`);
  if (!res.ok) throw new Error("Failed to get wellness scores");
  return res.json() as Promise<WellnessScoreResponse[]>;
}
